canary_tests.py

```diff
@@ -98,7 +98,9 @@ async def test_interrupts_and_pwm(self, pwm):
         should_stop = asyncio.Event()
         ticks = []
         tick_stream = await self.board.stream_ticks([interrupt])
-        counter_task = asyncio.create_task(self.record_tick_data(tick_stream, ticks, should_stop))
+
+        # if we don't return from record_tick_data in five seconds then we fail the test
+        counter_task = asyncio.create_task(asyncio.wait_for(self.record_tick_data(tick_stream, ticks, should_stop), 5))
```

Comment:
nit: the default Python style has a max line length of 80 characters. I'd put a newline just before `self`.